cachi2/core/package_managers/rubygems/main.py

```diff
@@ -0,0 +1,95 @@
+import logging
+from typing import Optional
+
+from cachi2.core.models.input import Request
+from cachi2.core.models.output import EnvironmentVariable, ProjectFile, RequestOutput
+from cachi2.core.models.sbom import Component
+from cachi2.core.package_managers.rubygems.parser import (
+    BundlerDependency,
+    GemDependency,
+    GitDependency,
+    PathDependency,
+)
+from cachi2.core.rooted_path import RootedPath
+
+log = logging.getLogger(__name__)
+
+
+def fetch_rubygems_source(request: Request) -> RequestOutput:
+    """Resolve and process all rubygems packages."""
+    components: list[Component] = []
+    environment_variables: list[EnvironmentVariable] = []
+    project_files: list[ProjectFile] = []
+
+    for package in request.packages:
+        path_within_root = request.source_dir.join_within_root(package.path)
+        _resolve_rubygems(package_dir=path_within_root, output_dir=request.output_dir)
+
+    return RequestOutput.from_obj_list(
+        components=components,
+        environment_variables=environment_variables,
+        project_files=project_files,
+    )
+
+
+def _resolve_rubygems(package_dir: RootedPath, output_dir: RootedPath) -> list[Component]:
+    """
+    Process a request for a single rubygems package.
+
+    TODO:
+    1. Parse Gemfile.lock
+    2. Get project metadata
+    3. Download dependencies
+    4. Generate SBOM components
+    """
```

Comment:
I rebased this branch on high-level design. This should only cover parsing Gemfil.lock. I will update it once high-level code is merged. It is marked as a draft because of that.

Reply:
Got it, thank you!